qupath-core/src/main/java/qupath/lib/images/servers/SlicedImageServer.java

```diff
@@ -0,0 +1,113 @@
+package qupath.lib.images.servers;
+
+import qupath.lib.io.GsonTools;
+import qupath.lib.regions.RegionRequest;
+
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * ImageServer that treats a particular set of z-slices and timepoints of another ImageServer
+ * as a full image.
+ */
+public class SlicedImageServer extends TransformingImageServer<BufferedImage> {
+
+    private final ImageServerMetadata metadata;
+    private final int zStart;
+    private final int zEnd;
+    private final int tStart;
+    private final int tEnd;
+
+    /**
+     * Create an ImageServer that represents a particular set of z-slices and timepoints of another ImageServer.
+     * <p>
+     * Index parameters of this function are not taken into account if their values is out of range (for example,
+     * a zStart of -1, or a tEnd of 10 with an image having only 5 timepoints).
+     *
+     * @param inputServer the input image to slice
+     * @param zStart the inclusive 0-based index of the first slice to consider
+     * @param zEnd the inclusive 0-based index of the last slide to consider
```

Comment:
Why not exclusive?
Making the end point exclusive feels more intuitive to me in Java, so that `0...nSlices` would give the entire image, rather than requiring `0...nSlices-1` to avoid being out of range.